core/tests/test_utils.py

```diff
@@ -196,6 +197,192 @@ def get_storage_model_classes():
                     yield clazz
 
 
+class ElasticSearchServicesStub(python_utils.OBJECT):
+    """The stub class that mocks the API functionality offered by the platform
+    layer, namely the platform.search elastic search services API.
+    """
+
+    _DB = {}
+
+    def add_documents_to_index(self, documents, index_name):
+        """Adds documents in a list of documents to a given index in the mock
+        database.
+
+        Args:
+            documents: list(dict). Each document should be a dictionary. Every
+                key in the document is a field name, and the corresponding
+                value will be the field's value. There MUST be a key named
+                'id', its value will be used as the document's id.
+            index_name: str. The name of the index to insert the document into.
+
+        Raises:
+            Exception. A document cannot be added to the index.
+        """
+        for document in documents:
+            assert 'id' in document
+            if index_name not in self._DB:
+                self._DB[index_name] = []
+            self._DB[index_name].append(document)
```

Comment:
Should this check that no other doc exists with that id?

More generally, would it be better for the internal structure of each index's value to be a dict keyed by doc_id? (Is there any advantage to having a list rather than a dict?) I guess one thing that this might help with is to ensure the stability of the sort for paged queries, in which case, perhaps explain that in the line comment for self._DB.